[BUG] {dot,}files in plugin directory cause startup failure

**Describe the bug**
While trying to package OpenSearch, I've come across the issue that first OpenSearch failed to start because of the plugin directory not being present as our packaging policy prevents installing empty directories.

```
[2021-04-28T10:41:13,181][ERROR][o.o.b.Bootstrap          ] [localhost] Exception
java.lang.IllegalStateException: Unable to access 'path.home' (/usr/share/opensearch/plugins)
[...]
```
Our usual way of solving this is creating a dotfile in the "empty" directories we need to install.
```
/usr/share/opensearch/plugins/.keep_web-apps_opensearch-0
```
which for example works fine for the data directory.

This results in OpenSearch failing to start with:
```
[2021-04-28T10:44:00,953][ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [localhost] uncaught exception in thread [main]
org.opensearch.bootstrap.StartupException: java.lang.IllegalStateException: Could not load plugin descriptor for plugin directory [.keep_web-apps_opensearch-0]
        at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:186) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:173) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:99) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:140) ~[opensearch-cli-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.cli.Command.main(Command.java:103) ~[opensearch-cli-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:139) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:105) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
Caused by: java.lang.IllegalStateException: Could not load plugin descriptor for plugin directory [.keep_web-apps_opensearch-0]
        at org.opensearch.plugins.PluginsService.readPluginBundle(PluginsService.java:420) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.plugins.PluginsService.findBundles(PluginsService.java:405) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.plugins.PluginsService.getPluginBundles(PluginsService.java:398) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:169) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.node.Node.<init>(Node.java:357) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.node.Node.<init>(Node.java:301) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:240) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:240) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:406) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:182) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        ... 6 more
Caused by: java.nio.file.FileSystemException: /usr/share/opensearch/plugins/.keep_web-apps_opensearch-0/plugin-descriptor.properties: Ist kein Verzeichnis
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:100) ~[?:?]
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106) ~[?:?]
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111) ~[?:?]
        at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:218) ~[?:?]
        at java.nio.file.Files.newByteChannel(Files.java:375) ~[?:?]
        at java.nio.file.Files.newByteChannel(Files.java:426) ~[?:?]
        at java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:420) ~[?:?]
        at java.nio.file.Files.newInputStream(Files.java:160) ~[?:?]
        at org.opensearch.plugins.PluginInfo.readFromProperties(PluginInfo.java:170) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.plugins.PluginsService.readPluginBundle(PluginsService.java:417) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.plugins.PluginsService.findBundles(PluginsService.java:405) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.plugins.PluginsService.getPluginBundles(PluginsService.java:398) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:169) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.node.Node.<init>(Node.java:357) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.node.Node.<init>(Node.java:301) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:240) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:240) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:406) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:182) ~[opensearch-1.0.0-beta1-SNAPSHOT.jar:1.0.0-beta1-SNAPSHOT]
        ... 6 more
```

**To Reproduce**
Steps to reproduce the behavior:
1. touch /usr/share/opensearch/plugins/.test
2. systemctl start opensearch.service
3. See error

**Expected behavior**
See OpenSearch starting successfully.

**Plugins**
None as there aren't any included in the opensearch-1.0.0-beta1-linux-x64.tar.gz tarball artifact

**Host/Environment (please complete the following information):**
 - OS: Linux
 - Version: opensearch-1.0.0-beta1

**Additional context**
Seeing there's actual code in the plugin directory in the repository https://github.com/opensearch-project/OpenSearch/tree/1.x/plugins is there any additional plugin artifact to fetch and install with opensearch by default? That way the plugin directory wouldn't be empty to begin with.

--- a/qa/no-bootstrap-tests/src/test/java/org/opensearch/bootstrap/SpawnerNoBootstrapTests.java
+++ b/qa/no-bootstrap-tests/src/test/java/org/opensearch/bootstrap/SpawnerNoBootstrapTests.java
@@ -215,33 +215,6 @@ public class SpawnerNoBootstrapTests extends LuceneTestCase {
                 equalTo("module [test_plugin] does not have permission to fork native controller"));
     }
 
-    public void testSpawnerHandlingOfDesktopServicesStoreFiles() throws IOException {
-        final Path esHome = createTempDir().resolve("home");
-        final Settings settings = Settings.builder().put(Environment.PATH_HOME_SETTING.getKey(), esHome.toString()).build();
-
-        final Environment environment = TestEnvironment.newEnvironment(settings);
-
-        Files.createDirectories(environment.modulesFile());
-        Files.createDirectories(environment.pluginsFile());
-
-        final Path desktopServicesStore = environment.modulesFile().resolve(".DS_Store");
-        Files.createFile(desktopServicesStore);
-
-        final Spawner spawner = new Spawner();
-        if (Constants.MAC_OS_X) {
-            // if the spawner were not skipping the Desktop Services Store files on macOS this would explode
-            spawner.spawnNativeControllers(environment, false);
-        } else {
-            // we do not ignore these files on non-macOS systems
-            final FileSystemException e = expectThrows(FileSystemException.class, () -> spawner.spawnNativeControllers(environment, false));
-            if (Constants.WINDOWS) {
-                assertThat(e, instanceOf(NoSuchFileException.class));
-            } else {
-                assertThat(e, hasToString(containsString("Not a directory")));
-            }
-        }
-    }
-
     private void createControllerProgram(final Path outputFile) throws IOException {
         final Path outputDir = outputFile.getParent();
         Files.createDirectories(outputDir);

--- a/server/src/main/java/org/opensearch/plugins/PluginsService.java
+++ b/server/src/main/java/org/opensearch/plugins/PluginsService.java
@@ -50,7 +50,6 @@ import org.opensearch.common.Strings;
 import org.opensearch.common.collect.Tuple;
 import org.opensearch.common.component.LifecycleComponent;
 import org.opensearch.common.inject.Module;
-import org.opensearch.common.io.FileSystemUtils;
 import org.opensearch.common.settings.Setting;
 import org.opensearch.common.settings.Setting.Property;
 import org.opensearch.common.settings.Settings;
@@ -365,7 +364,10 @@ public class PluginsService implements ReportingService<PluginsAndModules> {
         if (Files.exists(rootPath)) {
             try (DirectoryStream<Path> stream = Files.newDirectoryStream(rootPath)) {
                 for (Path plugin : stream) {
-                    if (FileSystemUtils.isDesktopServicesStore(plugin) || plugin.getFileName().toString().startsWith(".removing-")) {
+                    if (plugin.getFileName().toString().startsWith(".") && !Files.isDirectory(plugin)) {
+                        logger.warn(
+                            "Non-plugin file located in the plugins folder with the following name: [" + plugin.getFileName() + "]"
+                        );
                         continue;
                     }
                     if (seen.add(plugin.getFileName().toString()) == false) {

--- a/server/src/test/java/org/opensearch/plugins/PluginsServiceTests.java
+++ b/server/src/test/java/org/opensearch/plugins/PluginsServiceTests.java
@@ -32,7 +32,10 @@
 
 package org.opensearch.plugins;
 
+import org.opensearch.common.logging.Loggers;
 import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
 import org.apache.lucene.util.Constants;
 import org.apache.lucene.util.LuceneTestCase;
 import org.opensearch.LegacyESVersion;
@@ -44,6 +47,7 @@ import org.opensearch.common.settings.Settings;
 import org.opensearch.env.Environment;
 import org.opensearch.env.TestEnvironment;
 import org.opensearch.index.IndexModule;
+import org.opensearch.test.MockLogAppender;
 import org.opensearch.test.OpenSearchTestCase;
 import org.hamcrest.Matchers;
 
@@ -51,9 +55,7 @@ import java.io.IOException;
 import java.io.InputStream;
 import java.lang.reflect.InvocationTargetException;
 import java.net.URL;
-import java.nio.file.FileSystemException;
 import java.nio.file.Files;
-import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.Collection;
@@ -154,48 +156,15 @@ public class PluginsServiceTests extends OpenSearchTestCase {
         assertEquals(FilterablePlugin.class, scriptPlugins.get(0).getClass());
     }
 
-    public void testHiddenFiles() throws IOException {
+    public void testHiddenDirectories() throws IOException {
         final Path home = createTempDir();
         final Settings settings = Settings.builder().put(Environment.PATH_HOME_SETTING.getKey(), home).build();
         final Path hidden = home.resolve("plugins").resolve(".hidden");
         Files.createDirectories(hidden);
         @SuppressWarnings("unchecked")
         final IllegalStateException e = expectThrows(IllegalStateException.class, () -> newPluginsService(settings));
-
         final String expected = "Could not load plugin descriptor for plugin directory [.hidden]";
         assertThat(e, hasToString(containsString(expected)));
-    }
-
-    public void testDesktopServicesStoreFiles() throws IOException {
-        final Path home = createTempDir();
-        final Settings settings = Settings.builder().put(Environment.PATH_HOME_SETTING.getKey(), home).build();
-        final Path plugins = home.resolve("plugins");
-        Files.createDirectories(plugins);
-        final Path desktopServicesStore = plugins.resolve(".DS_Store");
-        Files.createFile(desktopServicesStore);
-        if (Constants.MAC_OS_X) {
-            @SuppressWarnings("unchecked")
-            final PluginsService pluginsService = newPluginsService(settings);
-            assertNotNull(pluginsService);
-        } else {
-            final IllegalStateException e = expectThrows(IllegalStateException.class, () -> newPluginsService(settings));
-            assertThat(e.getMessage(), containsString("Could not load plugin descriptor for plugin directory [.DS_Store]"));
-            assertNotNull(e.getCause());
-            assertThat(e.getCause(), instanceOf(FileSystemException.class));
-            if (Constants.WINDOWS) {
-                assertThat(e.getCause(), instanceOf(NoSuchFileException.class));
-            } else {
-                // force a "Not a directory" exception to be thrown so that we can extract the locale-dependent message
-                final String expected;
-                try (InputStream ignored = Files.newInputStream(desktopServicesStore.resolve("not-a-directory"))) {
-                    throw new AssertionError();
-                } catch (final FileSystemException inner) {
-                    // locale-dependent translation of "Not a directory"
-                    expected = inner.getReason();
-                }
-                assertThat(e.getCause(), hasToString(containsString(expected)));
-            }
-        }
     }
 
     public void testStartupWithRemovingMarker() throws IOException {
@@ -765,10 +734,25 @@ public class PluginsServiceTests extends OpenSearchTestCase {
         assertThat(e.getMessage(), containsString("my_plugin requires Java"));
     }
 
-    public void testFindPluginDirs() throws IOException {
+    public void testFindPluginDirs() throws Exception {
         final Path plugins = createTempDir();
 
+        final MockLogAppender mockLogAppender = new MockLogAppender();
+        mockLogAppender.start();
+        mockLogAppender.addExpectation(
+            new MockLogAppender.SeenEventExpectation(
+                "[.test] warning",
+                "org.opensearch.plugins.PluginsService",
+                Level.WARN,
+                "Non-plugin file located in the plugins folder with the following name: [.DS_Store]"
+            )
+        );
+        final Logger testLogger = LogManager.getLogger(PluginsService.class);
+        Loggers.addAppender(testLogger, mockLogAppender);
+
         final Path fake = plugins.resolve("fake");
+        Path testFile = plugins.resolve(".DS_Store");
+        Files.createFile(testFile);
 
         PluginTestUtil.writePluginProperties(
             fake,
@@ -791,6 +775,7 @@ public class PluginsServiceTests extends OpenSearchTestCase {
         }
 
         assertThat(PluginsService.findPluginDirs(plugins), containsInAnyOrder(fake));
+        mockLogAppender.assertAllExpectationsMatched();
     }
 
     public void testExistingMandatoryClasspathPlugin() {

